query/btccheckpoint.go

```diff
@@ -1,58 +1,72 @@
 package query
 
 import (
+	"context"
+
 	btcctypes "github.com/babylonchain/babylon/x/btccheckpoint/types"
-	"github.com/babylonchain/rpc-client/client"
+	"github.com/cosmos/cosmos-sdk/client"
 	sdkquerytypes "github.com/cosmos/cosmos-sdk/types/query"
 )
 
-// BTCCheckpointParams queries btccheckpoint module's parameters via ChainClient
-func BTCCheckpointParams(c *client.Client) (*btcctypes.Params, error) {
-	ctx, cancel := c.GetDefaultQueryContext()
+// QueryBTCCheckpoint queries the BTCCheckpoint module of the Babylon node
+// according to the given function
+func (c *QueryClient) QueryBTCCheckpoint(f func(ctx context.Context, queryClient btcctypes.QueryClient)) {
+	ctx, cancel := c.getQueryContext()
 	defer cancel()
 
-	queryClient := btcctypes.NewQueryClient(c.ChainClient)
-	req := &btcctypes.QueryParamsRequest{}
-	resp, err := queryClient.Params(ctx, req)
-	if err != nil {
-		return &btcctypes.Params{}, err
-	}
-	return &resp.Params, nil
-}
+	clientCtx := client.Context{Client: c.RPCClient}
+	queryClient := btcctypes.NewQueryClient(clientCtx)
 
-// BTCPositionAtEpoch queries btccheckpoint module for the Bitcoin position of an epoch
-func BTCPositionAtEpoch(c *client.Client, epochNumber uint64) (*btcctypes.QueryBtcCheckpointInfoResponse, error) {
-	ctx, cancel := c.GetDefaultQueryContext()
-	defer cancel()
+	f(ctx, queryClient)
+}
 
-	queryClient := btcctypes.NewQueryClient(c.ChainClient)
-	req := &btcctypes.QueryBtcCheckpointInfoRequest{
-		EpochNum: epochNumber,
-	}
+// BTCCheckpointParams queries btccheckpoint module's parameters via ChainClient
+func (c *QueryClient) BTCCheckpointParams() (*btcctypes.Params, error) {
+	var (
+		resp *btcctypes.QueryParamsResponse
+		err  error
+	)
+	c.QueryBTCCheckpoint(func(ctx context.Context, queryClient btcctypes.QueryClient) {
```

Comment:
e.g. for my comment above about returning errors from this function, this could be
```
err := c.QueryBTCCheckpoint(func(ctx context.Context, queryClient btcctypes.QueryClient) {
...
})
if err != nil {
    return nil, err
}
```

This way if the content of the function parameter had two kinds of errors, then the `BTCCheckpointParams` function would return the one propagated by the function.